flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/typeutils/AvroSerializer.java

```diff
@@ -99,9 +108,29 @@
 
 	/**
 	 * Creates a new AvroSerializer for the type indicated by the given class.
+	 * This constructor is intended to be used with {@link SpecificRecord} or reflection serializer.
+	 * For serializing {@link GenericData.Record} use {@link AvroSerializer#AvroSerializer(Class, Schema)}
 	 */
 	public AvroSerializer(Class<T> type) {
+		Preconditions.checkArgument(!isGenericRecord(type),
```

Comment:
Minor: Other preconditions checks in this class are done by statically imported methods. While this is not consistent within the code base, I would suggest to keep this consistent within a class as much as possible.